File: backend/src/main/java/br/ufpr/oscar/dto/LoginResponse.java
package br.ufpr.oscar.dto;

import br.ufpr.oscar.entity.Diretor;
import br.ufpr.oscar.entity.Filme;

public class LoginResponse {
    private boolean sucesso;
    private String mensagem;
    private Integer token;
    private boolean votoConfirmado;
    private Filme filme;
    private Diretor diretor;

    public LoginResponse(boolean sucesso, String mensagem, Integer token) {
        this(sucesso, mensagem, token, false, null, null);
    }

    public LoginResponse(boolean sucesso, String mensagem, Integer token,
                         boolean votoConfirmado, Filme filme, Diretor diretor) {
        this.sucesso = sucesso;
        this.mensagem = mensagem;
        this.token = token;
        this.votoConfirmado = votoConfirmado;
        this.filme = filme;
        this.diretor = diretor;
    }

    public boolean isSucesso()   { return sucesso; }
    public String getMensagem()  { return mensagem; }
    public Integer getToken()    { return token; }
    public boolean isVotoConfirmado() { return votoConfirmado; }
    public Filme getFilme() { return filme; }
    public Diretor getDiretor() { return diretor; }
}

File: mobile/app/src/main/java/br/ufpr/oscarmobile/model/LoginResponse.kt
package br.ufpr.oscarmobile.model
 
import com.google.gson.annotations.SerializedName

data class LoginResponse(
    @SerializedName("sucesso")         val sucesso: Boolean,
    @SerializedName("token")           val token: Int? = null,
    @SerializedName("mensagem")        val mensagem: String? = null,
    @SerializedName("votoConfirmado")  val votoConfirmado: Boolean = false,
    @SerializedName("filme")           val filme: Filme? = null,
    @SerializedName("diretor")         val diretor: Diretor? = null
)
